joft/base.py

```diff
@@ -236,6 +236,13 @@ def apply_reference_pool_to_payload(
             # TODO: write tests about replacing values and refactor this
             if not isinstance(v, (str, list)):
                 continue
+            # if the value is an simple value (string, int, float, bool) we skip it
+            if (
+                type(fields[field]) is int
```

Comment:
This is accessing only the fields. Why it's in the inner for loop?

Reply:
my bad